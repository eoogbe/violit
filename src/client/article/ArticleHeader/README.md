```jsx
const AUTHOR = 'fred-astaire';

const DATE_CREATED = '2019-09-29T01:23:45.679Z';

<ArticleHeader author={AUTHOR} dateCreated={DATE_CREATED} />
```
