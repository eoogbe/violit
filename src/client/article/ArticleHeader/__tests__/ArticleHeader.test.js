/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import TestRenderer from 'react-test-renderer';
import ArticleHeader from '../ArticleHeader';

const NOW = 1572566400000;

const AUTHOR = 'fred-astaire';

const DATE_CREATED = '2019-09-29T01:23:45.679Z';

it('renders without crashing', () => {
  Date.now = jest.fn().mockReturnValue(NOW);
  const renderer = TestRenderer.create(
    <ArticleHeader author={AUTHOR} dateCreated={DATE_CREATED} />
  );
  expect(renderer.toJSON()).toMatchSnapshot();
});
