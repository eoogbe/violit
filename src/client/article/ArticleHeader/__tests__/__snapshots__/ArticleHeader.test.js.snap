// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<header
  className="sc-gzVnrw iyjuRw"
>
  <span
    className="sc-htoDjs cMLlsL"
    itemProp="author"
  >
    fred-astaire
  </span>
  <time
    dateTime="2019-09-29T01:23:45.679Z"
    itemProp="dateCreated"
    title="2019-09-29T01:23:45.679Z"
  >
    1 month ago
  </time>
</header>
`;
